src/components/layout/Flex.js

```diff
@@ -1,33 +1,38 @@
-import omitProps from '@hocs/omit-props';
 import PropTypes from 'prop-types';
+import { createElement, useMemo } from 'react';
 import { View } from 'react-primitives';
-import { componentFromProp } from 'recompact';
-import styled from 'styled-components/primitives';
 
 export const getFlexStyleKeysFromShorthand = style =>
   style === 'end' || style === 'start' ? `flex-${style}` : style;
 
-const FlexPropBlacklist = [
-  'align',
-  'direction',
-  'flex',
-  'justify',
-  'self',
-  'wrap',
-];
-const FlexElement = omitProps(...FlexPropBlacklist)(
-  componentFromProp('component')
-);
+const Flex = ({
```

Comment:
Tbh I'm very uncertain if it's a good approach. Now we have nested two react components instead of one styled native primitive